src/Controllers/ShareDraftController.php

```diff
@@ -117,20 +117,59 @@ public function preview(HTTPRequest $request)
     /**
      * @param string $url
      *
+     * @deprecated Use {@link getRenderedPageByURL()}
+     *
      * @return HTTPResponse
      */
     protected function getRenderedPageByURLSegment($url)
     {
+        Deprecation::notice(
+            '3.0',
+            'getRenderedPageByURLSegment() should be replaced with getRenderedPageByURL()'
+        );
+
         $pageRequest = HTTPRequestBuilder::createFromEnvironment();
         $pageRequest->setHttpMethod('GET');
         $pageRequest->setUrl($url);
 
         $response = Director::singleton()->handleRequest($pageRequest);
+
         if ($response->isRedirect()) {
             // The redirect will probably be Absolute URL so just want the path
             $newUrl = parse_url($response->getHeader('location'), PHP_URL_PATH);
+
             return $this->getRenderedPageByURLSegment($newUrl);
         }
+
+        return $response;
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return HTTPResponse
+     */
+    private function getRenderedPageByURL($url)
```

Comment:
```suggestion
    private function getRenderedPageByURL(string $url): HTTPResponse
```

Up to top you'll also need to add
```
use SilverStripe\Control\HTTPResponse;
```